lib/miq_automation_engine/service_models/miq_ae_service_container_deployment.rb

```diff
@@ -0,0 +1,181 @@
+    end
+
+    def check_connection(deployment_host_ip, username, nodes_ips)
+      agent = LinuxAdmin::SSHAgent.new(ssh_auth.auth_key, "/tmp/ssh_manageiq/ssh_manageiq_#{id}")
+      success, unreachable_ips = agent.with_service do |socket|
+        ssh = LinuxAdmin::SSH.new(deployment_host_ip, username)
+        unreachable_ips = nodes_ips.select do |sub_ip|
+          !ssh.perform_commands(["sudo -E ssh -o 'StrictHostKeyChecking no' #{username}@#{sub_ip} echo $?"], socket)[:stdout].include?("0\r\n")
+        end
+        [unreachable_ips.empty?, unreachable_ips]
+      end
+      unless success
+        raise StandardError, "couldn't connect to : #{unreachable_ips.join(',')}"
+      end
+      success
+    end
+
+    def playbook_running?
+      pid = ar_method do
+        @object.customizations[:agent][:ansible_pid] if @object.customizations[:agent]
+      end
+      !pid.nil?
+    end
+
+    def run_playbook_command(ip, username, cmd)
+      result = {:finished => false}
+      ansible_pid = ar_method do
+        @object.customizations[:agent][:ansible_pid] if @object.customizations[:agent]
+      end
+      ssh = LinuxAdmin::SSH.new(ip, username, ssh_auth.auth_key)
+      if ansible_pid
+        process_running = ssh.perform_commands(["sudo kill -0 #{ansible_pid}"])[:stdout]
+        unless process_running.empty?
+          result[:finished] = true
+          result[:stdout] = ssh.perform_commands(["sudo cat /tmp/ansible.log"])[:stdout]
+          stop_agent(ssh)
+        end
+      else
+        pid, socket = create_agent(ssh, username)
+        ssh.perform_commands(["sudo rm -f /tmp/*ansible.log*"])
+        ssh.perform_commands(["SSH_AUTH_SOCK=#{socket} nohup sudo -b -E #{cmd}"])
+        ansible_pid = ssh.perform_commands(["sudo pgrep -f 'sudo -b -E'"])[:stdout].split("\r").first
+        customize(:ansible_pid => ansible_pid, :agent_pid => pid, :agent_socket => socket)
+      end
+      result
+    end
+
+    def agent_exists?
+      ar_method do
+        @object.customizations[:agent][:agent_pid] if @object.customizations[:agent]
+      end
+    end
+
+    def create_agent(ssh, username)
+      output = ssh.perform_commands(["ssh-agent -a ssh_manageiq_#{id}"])[:stdout]
+      socket = output.split('=')[1].split(' ')[0].chop
+      socket = "/#{username}/#{socket}"
+      socket = "/home#{socket}" if username != "root"
+      pid = output.split('=')[2].split(' ')[0].chop
+      ssh.perform_commands(["SSH_AUTH_SOCK=#{socket} SSH_AGENT_PID=#{pid} ssh-add -"], nil, ssh_auth.auth_key)
+      [pid, socket]
+    end
+
+    def stop_agent(ssh)
+      pid =  ar_method do
+        @object.customizations[:agent][:agent_pid]
+      end
+      socket = ar_method do
+        @object.customizations[:agent][:agent_socket]
+      end
+      ssh.perform_commands(["SSH_AGENT_PID=#{pid} ssh-agent -k &> /dev/null"])
+      ssh.perform_commands(["sudo rm -f #{socket}"])
+      customize(:ansible_pid => nil, :agent_pid => nil, :agent_socket => nil)
```

Comment:
It needs to be cleared because the process is done, there are some cases we run 2 playbooks in one deployment in case we deploy on rhel.